tools/notification-configuration/notification-creator/NotificationConfigurator.cs

```diff
@@ -7,29 +7,44 @@
 using Azure.Sdk.Tools.NotificationConfiguration.Enums;
 using Azure.Sdk.Tools.NotificationConfiguration.Models;
 using Azure.Sdk.Tools.NotificationConfiguration.Services;
-using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
+using Azure.Sdk.Tools.NotificationConfiguration.Helpers;
+using System;
+using Azure.Sdk.Tools.CodeOwnersParser;
+using System.Collections.Concurrent;
 
 namespace Azure.Sdk.Tools.NotificationConfiguration
 {
     class NotificationConfigurator
     {
         private readonly AzureDevOpsService service;
+        private readonly GitHubService gitHubService;
         private readonly ILogger<NotificationConfigurator> logger;
+        
+        private List<CodeOwnerEntry> codeOwnerEntries;
 
-        private const int MaxTeamNameLength = 64;
 
-        public NotificationConfigurator(AzureDevOpsService service, ILogger<NotificationConfigurator> logger)
+        private const int MaxTeamNameLength = 64;        
+        // Type 2 maps to a pipeline YAML file in the repository
+        private const int PipelineYamlProcessType = 2;
+        // A cache on the code owners github identity to owner descriptor.
+        private readonly Dictionary<string, string> codeOwnerCache = new Dictionary<string, string>();
```

Comment:
Try use the concurrent map in async call. However, we have to write more codes to handle exceptions. 
Kick a run for the async changes. 
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1503043&view=results

Reply:
Will have a follow up for further improvement. Do not want to introduce too many logic in this PR.